### ISSUE TITLE
Successfully cloning in Devfile, Error in the branch checkout (Authentication Required)

### ISSUE THREAD

**Issue by Deddinho23** [eclipse/che#22211]
### Describe the bug

For some reason, when we are trying to create a workspace from a devfile that have this structure:
```
projects:
- name: repo1
    git:
      remotes:
        origin: https://<<bitbucket-git-url>>
      checkoutFrom:
        revision: <<branch>>
  - name: repo2
    git:
      remotes:
        origin: https://<<bitbucket-git-url>>
      checkoutFrom:
        revision: <<branch>>
  - name: repo3
    git:
      remotes:
        origin: https://<<bitbucket-git-url>>
      checkoutFrom:
        revision: <<branch>>
```
The project-clone container fail the clone at the first repo (but the process continues and it would be created anyway), saying:
```
2023/05/12 14:27:57 Using temporary directory /projects/project-clone-1866860040
2023/05/12 14:27:57 Read DevWorkspace at /devworkspace-metadata/flattened.devworkspace.yaml
2023/05/12 14:27:57 Processing project repo1
2023/05/12 14:27:57 Cloning project repo1 to /projects/project-clone-1866860040/repo1
Cloning into '/projects/project-clone-1866860040/repo1'...
fatal: unable to get credential storage lock in 1000 ms: Read-only file system
2023/05/12 14:28:10 Cloned project repo1 to /projects/project-clone-1866860040/repo1
2023/05/12 14:28:10 Setting up remotes for project repo1
fatal: unable to get credential storage lock in 1000 ms: Read-only file system
2023/05/12 14:28:11 Fetched remote origin at https://<<bitbucket-git-url>>
2023/05/12 14:28:11 Found personal access token for URL https://<<bitbucket-git-url>>
2023/05/12 14:28:11 Encountered error while setting up project repo1: failed to checkout revision: failed to read remote origin: authentication required
```
But that's strange, because if it successfully cloned the repo, I dont see why it shouldn't complete the checkout of the branch..

PS: I noticed that every time you start the workspace, you have to do the "git config user.name" and "git config user.email" first if you want to commit some changes to the remote repo, maybe it can be the reason it is failing the checkout?

### Che version

7.66@latest

### Steps to reproduce

1. Create a workspace from a devfile that use "checkoutFrom" for 2 or more repos in the projects
2. Clone successfully the first repo, Fail to checkout: it doesn't clone the others
3. Workspace created with just one repo

### Expected behavior

1. Create a workspace from a devfile that use "checkoutFrom" for 2 or more repos in the projects
2. Clone and checkout successfully all the repo
3. Workspace created with all the repos from the projects of the devfile

### Runtime

Kubernetes (vanilla)

### Installation method

chectl/latest

### Environment

Amazon

### Additional context

We have the 7.66, but the problem started in our update from the 7.63 to 7.65

**Comment by ibuziuk** [comment]
@amisevsk @l0rd folks, could you please take a look?

**Comment by Deddinho23** [comment]
That's exactly the workaround we are using a the moment: instead of doing the checkout in the devfile, we've added it as a postStart event the exec of a script that do the checkout of the desired branch for every repo cloned, and it work normally

**Comment by Deddinho23** [comment]
Adding extra context that maybe can be useful: AutoProvision is set to false, so the creation of namespace and secret(PAT) is made directly by ourself ( following the doc about it in these pages: https://www.eclipse.org/che/docs/stable/administration-guide/provisioning-namespaces-in-advance/ https://www.eclipse.org/che/docs/stable/end-user-guide/using-a-git-provider-access-token/ )

The only thing that I found strange lately is that now you have to manually config git ("git config user.name | git config user.email") whenever you restart the workspace if you want to make some commit/push

**Comment by amisevsk** [comment]
> The only thing that I found strange lately is that now you have to manually config git ("git config user.name | git config user.email") whenever you restart the workspace if you want to make some commit/push, but I don't think it's related

I also don't think this is related. I believe the way that Che configures this automatically is by creating a secret in the workspace namespace:
```yaml
kind: Secret
apiVersion: v1
type: Opaque
metadata:
  name: user-profile
  namespace: <workspace-namespace>
  labels:
    app.kubernetes.io/part-of: che.eclipse.org
    controller.devfile.io/mount-to-devworkspace: 'true'
    controller.devfile.io/watch-secret: 'true'
  annotations:
    controller.devfile.io/mount-as: file
    controller.devfile.io/mount-path: /config/user/profile
data:
  email: <email for git>
  id: <I'm not sure...>
  name: <name>
```
(`data` is base64-encoded, use `stringData` when creating to avoid some hassle)

The PAT being used should be stored in a secret named `devworkspace-merged-git-credentials` in the workspace. Could you check for this secret in the workspace's namespace, and verify that its contents follow the expected format?
```
data:
  credentials: https://{USERNAME}:{PERSONAL_ACCESS_TOKEN}@{GIT_WEBSITE}
```
(Don't share these contents here, they're credentials :))

I've created an issue to at least not stop after hitting an error cloning one of multiple projects: https://github.com/devfile/devworkspace-operator/issues/1101

**Comment by Deddinho23** [comment]
Thank you for the feedback @amisevsk!

About the problem with te git config, the "user-profile" secret was auto-provisioned correctly, but I think I found why its keep asking for the manual config when trying to push:
![image](https://github.com/eclipse/che/assets/48152430/4cfeb8cf-00e1-4c56-bb04-66a705029d1c)
_(I deleted the workspace after the test, so that ws-id its not sensitive)_
It seems like it didn't "feel" the config stored in the secret mounted on "/config/user/profile" path, but only the local ones

Meanwhile I checked about the devworkspace-merged-git-credentials and the pattern of the credentials is correct... we are using an hosted Bitbucket Server, idk if that can help for replicate the problem

**Comment by Deddinho23** [comment]
After the last update (from 7.66 to 7.70), we are no longer face this problem! 😃

**Comment by amisevsk** [comment]
Glad to hear it @Deddinho23! Apologies for missing your last comment -- I think your issue was actually https://github.com/devfile/devworkspace-operator/issues/1114 (we weren't using the configured PAT when checking out a revision). This was fixed in DevWorkspace Operator v0.21.1 (PR: https://github.com/devfile/devworkspace-operator/pull/1115)

I'll close this issue for now, but let me know if you think it should be reopened.
